testnet/stacks-node/src/nakamoto_node/signer_coordinator.rs

```diff
@@ -293,19 +301,52 @@ impl SignerCoordinator {
         sortdb: &SortitionDB,
         counters: &Counters,
     ) -> Result<Vec<MessageSignature>, NakamotoNodeError> {
+        // build a BTreeMap of the various timeout steps
+        let mut block_rejection_timeout_steps = BTreeMap::<u64, Duration>::new();
+        for (percentage, duration) in self.block_rejection_timeout_steps.iter() {
+            let rejections_amount =
+                ((self.total_weight as f64 / 100.0) * *percentage as f64) as u64;
```

Comment:
```suggestion
                ((f64::from(self.total_weight)/ 100.0) * f64::from(percentage)) as u64;
```

I suggest changing the type of the `block_rejection_timeout_steps` hashmap to `<u32, Duration>` so that f64 has a lossless conversion (`from` is also preferable to `as`). That isn't possible for the conversion back to `u64`, so that has to remain `as u64`.

Reply:
moved the HashMap key to u32